pkg/apis/pingcap/v1alpha1/backup.go

```diff
@@ -187,3 +222,77 @@ func IsCleanCandidate(backup *Backup) bool {
 func NeedNotClean(backup *Backup) bool {
 	return backup.Spec.CleanPolicy == CleanPolicyTypeOnFailure && !IsBackupFailed(backup)
 }
+
+// ParseLogBackupSubCommand parse the log backup subcommand from cr.
+// The parse priority of the command is stop > truncate > start.
+func ParseLogBackupSubcommand(backup *Backup) LogSubCommandType {
+	if backup.Spec.Mode != BackupModeLog {
+		return ""
+	}
+	if backup.Spec.LogStop {
+		return LogStopCommand
+	}
+	if backup.Spec.LogTruncateUntil != "" {
+		return LogTruncateCommand
+	}
+	return LogStartCommand
+}
+
+// IsLogBackupSubCommandOntheCondition return whether the log subcommand on the condition.
+func IsLogBackupSubCommandOntheCondition(backup *Backup, conditionType BackupConditionType) bool {
+	command := ParseLogBackupSubcommand(backup)
+	switch command {
+	case LogStartCommand, LogStopCommand:
+		if subStatus, ok := backup.Status.LogSubCommandStatuses[command]; ok {
+			return subStatus.Phase == conditionType
+		}
+	case LogTruncateCommand:
+		// truncate Command's truncating until is the spec truncate until means the truncate is in progress.
+		if subStatus, ok := backup.Status.LogSubCommandStatuses[command]; ok {
+			return subStatus.LogTruncatingUntil == backup.Spec.LogTruncateUntil && subStatus.Phase == conditionType
+		}
+	default:
+		return false
+	}
+	return false
+}
+
+// GetLogSumcommandConditionInfo gets log subcommand current phase's reason and message
+func GetLogSumcommandConditionInfo(backup *Backup) (reason, message string) {
```

Comment:
Maybe the function name should be `GetLogSubcommandConditionInfo`?

Reply:
yes